### PR TITLE
feat: exclude bots to assign experiments

### PR DESCRIPTION
**Pull request by hecktarzuli**
The exclude bot feature.
Setup simple tests.

### CODE REVIEW

File: lib/plugin.js [hunk 1]
```diff
@@ -1,10 +1,8 @@
 import weightedRandom from 'weighted-random'
 import { parse as parseCookie, serialize as serializeCookie } from 'cookie'
-
 import experiments from '<%= options.experimentsDir %>'
-const MAX_AGE = <%= options.maxAge %>
 
-export default function (ctx, inject) {
+export default function (ctx, inject) {
```
Comment (pi0) [inline lib/plugin.js:5]: extra spaces

File: lib/plugin.js [hunk 2]
```diff
@@ -130,3 +129,18 @@ function googleOptimize({ experiment }) {
 
   window.ga('set', 'exp', exp)
 }
+
+// should we skip bots?
+function skipAssignment(ctx) {  
+  if(!<%= options.excludeBots %>) return false
```
Comment (pi0) [inline lib/plugin.js:135]: Neat: space before `(` and please use `{ }` for consistency

File: package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "nuxt-google-optimize",
-  "version": "0.5.4",
+  "version": "0.5.5",
```
Comment (pi0) [inline package.json:3]: We shouldn't bump version for PR but release script from master will bump and update changelog

File: package.json [hunk 4]
```diff
@@ -17,7 +20,7 @@
     "access": "public"
   },
   "scripts": {
-    "dev": "nuxt test/fixture",
+    "dev": "nuxt test/fixture/bot-block",
```
Comment (pi0) [inline package.json:23]: Why changing fixtures to a specific feature? Can't we add this to the main fixture to skip bots?

File: test/bot-default.test.js [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+  test('bot bot', async () => {
+    await page.setUserAgent('AdsBot-Google (+http://www.google.com/adsbot.html)')
+    await page.goto(url('/'))
+
+    const $exp = await page.evaluate(() => window.$exp)
+    expect($exp.name).toBe('test1')
+  })
+
+  test('bot spider', async () => {
```
Comment (pi0) [inline test/bot-default.test.js:27]: We may generate tests using an array as they are identical

```js
const botUserAgents = [...]

for (const agent of botUserAgents) {
  test('', () =>)
}
```

File: lib/module.js [hunk 6]
```diff
@@ -6,8 +6,11 @@ module.exports = async function module (moduleOptions) {
       pushPlugin: true,
       experimentsDir: '~/experiments',
       maxAge: 60 * 60 * 24 * 7, // 1 Week
-      plugins: []
+      plugins: [],
+      excludeBots: false,
```
Comment (pi0) [inline lib/module.js:10]: I think it would be nice and non breaking change if we enable it by default

File: lib/plugin.js (hunk 2 above)
Comment (hecktarzuli) [reply lib/plugin.js:135]: Space - I'm surprised the aggressive eslint rules didn't catch this.
IMHO all bool functions should always return bool, not something that's falsy, but I can change it.

File: package.json (hunk 4 above)
Comment (hecktarzuli) [reply package.json:23]: If you want it to block by default, sure I can change this to 'basic'.  I didn't think it mattered that much (shrug)

File: lib/plugin.js (hunk 2 above)
Comment (pi0) [reply lib/plugin.js:135]: Indeed long time issue with lodash templates that are not lintable. There is [eslint-multiplexer](https://github.com/pimlie/eslint-multiplexer) ([usage](https://github.com/nuxt/nuxt.js/blob/dev/package.json#L23)) but not easy to use. Working on a more linter friendly template engine.

File: test/module.test.js [hunk 7]
```diff
@@ -1,26 +1,84 @@
...
+      await page.setUserAgent(agent)
+      await page.goto(url('/'))
+
+      const $exp = await page.evaluate(() => window.$exp)
+      expect($exp.name).toEqual(undefined)
+    })
+  }
+
+  const unBlockedUserAgents = [
+    'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/74.0.3729.169 Safari/537.36',
```
Comment (hecktarzuli) [inline test/module.test.js:71]: I could consolidate this more into a 'agentKey': result, but thought that would complicate the dev experience a bit much (shrug)

File: test/module.test.js [hunk 8]
```diff
@@ -1,26 +1,84 @@
...
+    ''
+  ]
+
+  for (const agent of unBlockedUserAgents) {
+    test('', async () => {
```
Comment (pi0) [inline test/module.test.js:76]: Neat: First argument may be agent too 🙈

Comment (hecktarzuli) [reply test/module.test.js]: Done

**Pull request merged**